user can choose cpu difficulty

Cristian-Baeza opened the pull request.

td-extreme commented on lib/game_loop.rb:
```diff
@@ -27,7 +27,11 @@ def core_game
         @game_logic.take_turn(user_choice)
       else
-        @computer_logic.cpu_turn
+        if @game_config.cpu_difficulty == :easy
+          @computer_logic.cpu_turn_easy
+        else
+          @computer_logic.cpu_turn_hard
+        end
```
There is no reason that the game loop should be modified for this change.

Cristian-Baeza replied:
Got it! It was a doozy adjusting test specs to pass but figured it out

td-extreme reviewed (changes requested):
Can you complete this without modifying the game_loop.

Cristian-Baeza commented:
> Can you complete this without modifying the game_loop?

Got a working solution that does not change game_loop. 
-Changed game_loop back to the way it was 
- Computer_logic now uses game_config to keep track of game difficulty
- Adjusted tests as needed for them to pass based on the changes made, mainly computer_logic now needing game_config during initialize

td-extreme commented:
Per our stand up talk -
* Let's change the `ComputerLogic` class into `ComputerLogicHard` and create a new `ComputerLogicEasy` class and test file.
* Then have the main bin/tictactoe.rb script decide which class to pass to the game_loop
* While were at it can we put the computer players into their own folder to organize things a little better
`/lib/computer_logic/computer_logic_easy.rb`
`/lib/computer_logic/computer_logic_hard.rb`
